server/src/main/java/org/apache/druid/server/AsyncQueryForwardingServlet.java

```diff
@@ -449,6 +463,95 @@ static String getAvaticaConnectionId(Map<String, Object> requestMap)
     return (String) connectionIdObj;
   }
 
+  static String getAvaticaProtobufConnectionId(Service.Request request)
+  {
+    if (request instanceof Service.CatalogsRequest) {
```

Comment:
These `if` clause seems fragile from changes on Avatica request types (such as new request type), but I'm not sure if there is a better way..

Reply:
Agreed, it's a bit fragile, however, that, I'm afraid, is the nature of protobufs. I'm not sure how likely they are to add new request types without significant overhaul to Calcite though